dataengineeringutils3/s3.py

```diff
@@ -215,3 +217,104 @@ def write_local_file_to_s3(local_file_path, s3_path, overwrite=False):
         resp = s3_resource.meta.client.upload_file(local_file_path, bucket, key)
 
     return resp
+
+
+def write_local_folder_to_s3(
+    root_folder: Union[Path, str],
+    s3_path: str,
+    overwrite: bool = False,
+    include_hidden_files: bool = True,
+    current_folder: Union[Path, str] = None,
+) -> None:
+    """Copy a local folder and all its contents to s3, keeping its directory structure.
+
+    :param root_folder: the folder whose contents you want to upload
+    :param s3_path: where you want the folder to be located when it's uploaded
+    :param overwrite: if True, overwrite existing files in the target location
+        if False, raise ValueError if existing files are found in the target location
+    :param include_hidden_files: if False, ignore files whose names start with a .
+    :param current_folder: leave as None - only used during recursion
```

Comment:
If this should be left as `None` why is it an option?

Reply:
You know what? I'm a total wally. The satisfaction of doing it recursively blinded me to a much simpler approach that removes the need for this extra parameter and should run more efficiently too. Now just uses `Path.rglob()` rather than iterating through each directory.